app/scripts/controllers/metametrics.js

```diff
@@ -479,13 +479,33 @@ export default class MetaMetricsController {
     }
   }
 
+  // Method to track metrics event
+  // If an actionId is provided createEventFragment, finalizeEventFragment are used to track event,
+  // this is to achieve idempotent behaviour.
+  // If actionid is not provided this._trackEvent is used to track the event.
+  trackEvent(payload, options) {
+    if (options?.actionId) {
+      const fragment = this.createEventFragment({
+        ...payload,
+        ...options,
+        actionId: options?.actionId,
```

Comment:
[nit] 
```suggestion
        actionId: options.actionId,
```

`options` has to be defined to get here